lib/AST/ModuleLoader.cpp

```diff
@@ -254,7 +254,7 @@ ModuleDependencyInfo::collectCrossImportOverlayNames(
       ModuleDecl::collectCrossImportOverlay(ctx, file, moduleName,
                                             bystandingModule);
     result[bystandingModule] = std::move(overlayNames);
-    overlayFiles.push_back({moduleName.str(), file.str()});
+    overlayFiles.insert({moduleName.str(), file.str()});
```

Comment:
`collectCrossImportOverlay` is not exactly cheap. Can we skip if we know the entry is in the set already?